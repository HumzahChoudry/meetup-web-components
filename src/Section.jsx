import React from 'react';
import cx from 'classnames';

/**
 * @module Section
 */
class Section extends React.Component {
	render() {
		const {
			children,
			className,
			noBorder,
			...other
		} = this.props;

		const classNames = cx(
			'section',
			{
				noBorder: noBorder
			},
			className
		);

		return (
			<div
				className={classNames}
				{...other}>
					{children}
			</div>
		);
	}
}

Section.propTypes = {
	noBorder: React.PropTypes.bool
};

export default Section;
