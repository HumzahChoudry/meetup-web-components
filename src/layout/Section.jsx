import React from 'react';
import cx from 'classnames';

export const SECTION_CLASS = 'section';
export const SECTION_NOBORDER_CLASS = 'section--noBorder';
export const VALID_BREAKPOINTS = {
	all: 'atAll',
	medium: 'atMedium',
	large: 'atLarge',
};

/**
 * Design System Component: Provides `section` container for components
 * @module Section
 */
class Section extends React.Component {
	render() {
		const {
			children,
			className,
			noBorder,
			...other
		} = this.props;

		const noBorderBreakpoint = VALID_BREAKPOINTS[noBorder] || VALID_BREAKPOINTS['all'];

		const classNames = cx(
			SECTION_CLASS,
			{
				[`${noBorderBreakpoint}_${SECTION_NOBORDER_CLASS}`]: noBorder,
			},
			className
		);

		return (
			<section
				className={classNames}
				{...other}
			>
				{children}
			</section>
		);
	}
}
Section.propTypes = {
	noBorder: React.PropTypes.oneOf(Object.keys(VALID_BREAKPOINTS))
};

export default Section;
