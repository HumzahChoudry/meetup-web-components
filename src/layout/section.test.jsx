import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-addons-test-utils';

import Section, {
	SECTION_CLASS,
	SECTION_NOBORDER_CLASS,
	VALID_BREAKPOINTS,
} from './Section';

describe('Section', function() {
	let section,
		sectionNode;

	beforeEach(() => {
		section = TestUtils.renderIntoDocument(<Section />);
		sectionNode = ReactDOM.findDOMNode(section);
	});
	afterEach(() => {
		section = null;
		sectionNode = null;
	});
	it('exists', function() {
		expect(sectionNode).not.toBeNull();
	});
	it(`check that default component has '${SECTION_CLASS}' class`, function() {
		expect(sectionNode.classList).toContain(SECTION_CLASS);
	});

	describe('Section noBorder', () => {
		it(`check that default component has '${SECTION_NOBORDER_CLASS}' class`, function() {
			Object.keys(VALID_BREAKPOINTS).forEach(breakpoint => {
				section = TestUtils.renderIntoDocument(<Section noBorder={breakpoint} />);
				sectionNode = ReactDOM.findDOMNode(section);
				expect(sectionNode.classList).toContain(SECTION_CLASS);
				expect(sectionNode.classList).toContain(`${VALID_BREAKPOINTS[breakpoint]}_${SECTION_NOBORDER_CLASS}`);
			});
		});
	});
});
