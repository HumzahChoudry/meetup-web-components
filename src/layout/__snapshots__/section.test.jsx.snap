// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section exists 1`] = `
<section
  className="section"
/>
`;
